python/lsst/afw/geom/skyWcs/skyWcsContinued.py

```diff
@@ -20,9 +20,79 @@
 # see <http://www.lsstcorp.org/LegalNotices/>.
 #
 
-__all__ = []
+import numpy as np
 
+from lsst.utils import continueClass
 from ..python import reduceTransform
 from .skyWcs import SkyWcs
 
+__all__ = []
+
+
+@continueClass  # noqa: F811
+class SkyWcs:
+    def pixelToSkyArray(self, x, y, degrees=False):
+        """
+        Convert numpy array pixels (x, y) to numpy array sky (ra, dec)
+        positions.
+
+        Parameters
+        ----------
+        x : `np.ndarray`
+            Array of x values.
+        y : `np.ndarray`
+            Array of y values.
+        degrees : `bool`, optional
+            Return ra, dec arrays in degrees if True.
+
+        Returns
+        -------
+        ra : `np.ndarray`
+            Array of Right Ascension.  Units are radians unless
+            degrees=True.
+        dec : `np.ndarray`
+            Array of Declination.  Units are radians unless
+            degrees=True.
+        """
+        xy = np.vstack((x, y))
+        ra, dec = np.vsplit(self.getTransform().getMapping().applyForward(xy), 2)
+        ra %= (2.*np.pi)
+
+        if degrees:
+            return np.rad2deg(ra.ravel()), np.rad2deg(dec.ravel())
```

Comment:
I don't think `ravel` is useful here. If `xy` does not have exactly two rows, the call to `applyForward` will throw an error. And when `xy` has exactly two rows, `ra` and `dec` will be flat. Flattening isn't incorrect, but may just add to unnecessary calls without actually covering any edge cases.

Reply:
Unfortunately, it seems that `vsplit` does not return 1d arrays but returns 2 2d arrays with shape `[1, npoints]`.  The `ravel` call will flatten that without copying memory (unlike `flatten` which does a copy).  In addition, a `ravel` on a true 1d array is essentially a no-op.